src/qhttp/Response.cc

```diff
@@ -186,23 +209,49 @@ std::string Response::_headers() const {
     return headerst.str();
 }
 
-void Response::_write() {
+void Response::_startTransmit() {
     if (_transmissionStarted.test_and_set()) {
         LOGLS_ERROR(_log, logger(_server)
                                   << logger(_socket) << "handler logic error: multiple responses sent");
-        return;
     }
+}
+
+void Response::_finishTransmit(boost::system::error_code const& ec, std::size_t sent) {
+    if (ec) {
+        LOGLS_ERROR(_log, logger(_server) << logger(_socket) << "write failed: " << ec.message());
+    }
+    _responseBuf.clear();
```

Comment:
Would an error/info log if we haven't sent the full file (e.g. `_bytesToRead != 0`) be a good idea?

Reply:
The method `_finishTransmit` gets called from 4 locations when `_bytesToRead == 0` or when `ec` is not a success. So, the extra check inside the method is not needed.

Reply:
These are invocations of the method mentioned above:
```
void Response::send(std::string const& content, std::string const& contentType) {
    ...
    asio::async_write(*_socket, asio::buffer(_responseBuf.data(), _responseBuf.size()),
                      [self = shared_from_this()](boost::system::error_code const& ec, std::size_t sent) {
                          self->_finishTransmit(ec, sent);
                      });
}
```
In that case `_bytesToRead` is always set to `0` in the constructor.

The following two cases are when something bad happens:
```
auto const ec = boost::system::system_error(errno, boost::system::generic_category());
_finishTransmit(ec.code(), _bytesSent);
```
```
if (ec) {
    self->_finishTransmit(ec, self->_bytesSent);
```

The last case is when the transfer is over:
```
if (self->_bytesRemaining == 0) {
    auto const ec = boost::system::errc::make_error_code(boost::system::errc::success);
    self->_finishTransmit(ec, self->_bytesSent);

```